File: backend/src/main/java/zw/co/hisolutions/core/service/TopicService.java
package zw.co.hisolutions.core.service;
  
import zw.co.hisolutions.common.service.GenericService;
import zw.co.hisolutions.core.entity.Topic;

public interface TopicService extends GenericService<Topic, Long>{
    public Topic findByName(String name);   
}

File: frontend/src/app/admin/services/rest/topic.service.ts
import { Injectable } from '@angular/core'; 
import {HttpClient} from '@angular/common/http';
import {DataService} from 'shared/services/data.service'; 

@Injectable()
export class TopicService extends DataService {

    constructor(httpClient: HttpClient) {
        const resourcePath =  '/rest/topics'; 
        super(httpClient, resourcePath);
    }

}
